exp/addons/controllers/clusterresourceset_controller.go

```diff
@@ -84,7 +84,6 @@ func (r *ClusterResourceSetReconciler) SetupWithManager(ctx context.Context, mgr
 		Watches(
 			&source.Kind{Type: &corev1.Secret{}},
 			handler.EnqueueRequestsFromMapFunc(r.resourceToClusterResourceSet),
-			builder.OnlyMetadata,
```

Comment:
The reason to cache only metadata was to avoid storing all secrets of a Cluster in memory, which can be quite intensive. Is there any alternative we can think here? Could we do the check in a different part of the codebase?

Reply:
Yes this is not prefered but the only other way I could think of is reconciling for all secrets created like we do for ConfigMaps and return early in the mapping function.

Reply:
Let's drop the check for the `Type` in the predicate and let the controller figure out if it should retrieve the secret or not?

Reply:
https://github.com/kubernetes-sigs/cluster-api/blob/master/exp/addons/controllers/predicates/resource_predicates.go#L50-L55

Reply:
Done.